release.nix

```diff
@@ -188,27 +198,61 @@ let
       # grpc-haskell{-,core} code into `ghci` from within `nix-shell`
       # environments.
       #
-      # TODO: We might try using pkgsNew.fixDarwinDylibNames (see PR#129)
+      # TODO: We might try using pkgsSelf.fixDarwinDylibNames (see PR#129)
       # instead of setting DYLD_LIBRARY_PATH, but we might still need them
       # around for `ghci` as on Linux.
 
-      pkgsNew.haskell.lib.overrideCabal haskellPackage (oldAttributes: {
-          preBuild = (oldAttributes.preBuild or "") +
-            pkgsNew.lib.optionalString pkgsNew.stdenv.isDarwin ''
-              export DYLD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH
-            '';
+      pkgsSelf.haskell.lib.overrideCabal haskellPackage (oldAttributes: {
+        preBuild = (oldAttributes.preBuild or "") +
+          pkgsSelf.lib.optionalString pkgsSelf.stdenv.isDarwin ''
+            export DYLD_LIBRARY_PATH=${pkgsSelf.grpc}/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH
+          '';
 
-          shellHook = (oldAttributes.shellHook or "") +
-            pkgsNew.lib.optionalString pkgsNew.stdenv.isDarwin ''
-              export DYLD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH
-            '' +
-            pkgsNew.lib.optionalString pkgsNew.stdenv.isLinux ''
-              export LD_LIBRARY_PATH=${pkgsNew.grpc}/lib''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
-            '';
-        }
-      );
+        shellHook = (oldAttributes.shellHook or "") +
+          pkgsSelf.lib.optionalString pkgsSelf.stdenv.isDarwin ''
+            export DYLD_LIBRARY_PATH=${pkgsSelf.grpc}/lib''${DYLD_LIBRARY_PATH:+:}$DYLD_LIBRARY_PATH
+          '' +
+          pkgsSelf.lib.optionalString pkgsSelf.stdenv.isLinux ''
+            export LD_LIBRARY_PATH=${pkgsSelf.grpc}/lib''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
+          '';
+      });
   };
 
+  hsAddPatch = pkgs: patchFile:
+    pkgs.lib.flip pkgs.haskell.lib.overrideCabal (old: {
+      patches = (old.patches or [ ]) ++ [ patchFile ];
+    });
```

Comment:
You can use `pkgs.haskell.lib.compose.appendPatch`

Reply:
@Gabriella439 eliminated this function in favor of `pkgs.haskell.lib.compose.appendPatch`